comp/rdnsquerier/impl/config_test.go

```diff
@@ -0,0 +1,184 @@
+network_devices:
+  netflow:
+    reverse_dns_enrichment_enabled: true
+reverse_dns_enrichment:
+  workers: 0
+  chan_size: 0
+  rate_limiter:
+    enabled: true
+    limit_per_sec: 0
+  cache:
+    enabled: true
+    entry_ttl: 0
+    clean_interval: 0
+    persist_interval: 0
+`,
+			expectedConfig: rdnsQuerierConfig{
+				enabled:              true,
+				workers:              defaultWorkers,
+				chanSize:             defaultChanSize,
+				rateLimiterEnabled:   true,
+				rateLimitPerSec:      defaultRateLimitPerSec,
+				cacheEnabled:         true,
+				cacheEntryTTL:        defaultCacheEntryTTL,
+				cacheCleanInterval:   defaultCacheCleanInterval,
+				cachePersistInterval: defaultCachePersistInterval,
+			},
+		},
+		{
+			name: "specific config",
+			configYaml: `
+network_devices:
+  netflow:
+    reverse_dns_enrichment_enabled: true
+reverse_dns_enrichment:
+  workers: 25
+  chan_size: 999
+  rate_limiter:
+    enabled: true
+    limit_per_sec: 111
+  cache:
+    enabled: true
+    entry_ttl: 24h
+    clean_interval: 30m
+    persist_interval: 2h
+`,
+			expectedConfig: rdnsQuerierConfig{
+				enabled:              true,
+				workers:              25,
+				chanSize:             999,
+				rateLimiterEnabled:   true,
+				rateLimitPerSec:      111,
+				cacheEnabled:         true,
+				cacheEntryTTL:        24 * time.Hour,
+				cacheCleanInterval:   30 * time.Minute,
+				cachePersistInterval: 2 * time.Hour,
+			},
+		},
+		{
+			name: "specific config with defaults when rate limiter and cache are enabled",
+			configYaml: `
+network_devices:
+  netflow:
+    reverse_dns_enrichment_enabled: true
+reverse_dns_enrichment:
+  workers: 25
+  chan_size: 999
+  rate_limiter:
+    enabled: true
+  cache:
+    enabled: true
+`,
+			expectedConfig: rdnsQuerierConfig{
+				enabled:              true,
+				workers:              25,
+				chanSize:             999,
+				rateLimiterEnabled:   true,
+				rateLimitPerSec:      defaultRateLimitPerSec,
+				cacheEnabled:         true,
+				cacheEntryTTL:        defaultCacheEntryTTL,
+				cacheCleanInterval:   defaultCacheCleanInterval,
+				cachePersistInterval: defaultCachePersistInterval,
+			},
+		},
+		{
+			name: "specific config with rate limiter and cache disabled",
+			configYaml: `
+network_devices:
+  netflow:
+    reverse_dns_enrichment_enabled: true
+reverse_dns_enrichment:
+  workers: 25
+  chan_size: 999
+  rate_limiter:
+    enabled: false
+  cache:
+    enabled: false
+`,
+			expectedConfig: rdnsQuerierConfig{
+				enabled:              true,
+				workers:              25,
+				chanSize:             999,
+				rateLimiterEnabled:   false,
+				rateLimitPerSec:      0,
+				cacheEnabled:         false,
+				cacheEntryTTL:        0,
+				cacheCleanInterval:   0,
+				cachePersistInterval: 0,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config.Datadog().SetConfigType("yaml")
+			err := config.Datadog().ReadConfig(strings.NewReader(tt.configYaml))
```

Comment:
I think we have a potential issue with the way we are setting up the configuration. Since we are using the global config object in the test there could be a potential for causing issues on other parts of the code. 

We have a mock object that you could use to create a config object just for your tests and would take care for cleanup for you `pkg/config/mock.go`

Could we use it for this tests ?

Reply:
Changed to use ConfFromYAML() in commit 04687a25587bb09f3fe0a7e2fdfe8ead0279e98c.  Let me know if that's suitable or if you'd prefer a different approach.